### PR TITLE
Added zwave device to product database: ZWave.me blinds controller

### PR DESCRIPTION
**Pull request by mleegwt**
To support a Zwave device that I'm using to control blinds I added the product to the database.

Best regards,

Michiel

### CODE REVIEW

File: bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java [hunk 1]
```diff
@@ -321,30 +322,10 @@ private void handleMultiChannelCapabilityReportResponse(SerialMessage serialMess
...
-			Basic basic = this.getNode().getDeviceClass().getBasicDeviceClass();
-			Generic generic = Generic.getGeneric(genericDeviceClass);
-			if (generic == null) {
-				logger.error(String.format("NODE %d: Endpoint %d has invalid device class. generic = 0x%02x, specific = 0x%02x.", 
-						this.getNode().getNodeId(), endpoint, genericDeviceClass, specificDeviceClass));
+			if (!updateDeviceClass(endpoint, genericDeviceClass, specificDeviceClass, dynamic)) {
+				// Updating device class failed, already logged, continue with next endpoint
 				continue;
 			}
-			Specific specific = Specific.getSpecific(generic, specificDeviceClass);
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java]: Why have you deleted all this code?

File: bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveConfigurationCommandClass.java [hunk 2]
```diff
@@ -13,14 +13,13 @@
 
 import org.openhab.binding.zwave.internal.protocol.ConfigurationParameter;
 import org.openhab.binding.zwave.internal.protocol.SerialMessage;
-import org.openhab.binding.zwave.internal.protocol.ZWaveController;
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveConfigurationCommandClass.java]: Why have you changed this file? It doesn't look like it's needed?

File: bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java [hunk 3]
```diff
@@ -353,34 +334,7 @@ private void handleMultiChannelCapabilityReportResponse(SerialMessage serialMess
 			}
 
 			// Add all the command classes supported by this endpoint
-			for (int i = 0; i < serialMessage.getMessagePayload().length - offset - 3; i++) {
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java]: Why have you deleted this code?

Comment (mleegwt) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java]: The code is moved (extracted). To improve readability. No functional change in this commit.

File: bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveConfigurationCommandClass.java (hunk 2 above)
Comment (mleegwt) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveConfigurationCommandClass.java]: It was an unused import, a warning was presented in Eclipse.

File: bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java (hunk 1 above)
Comment (mleegwt) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java]: The code is moved (extracted). To improve readability. No functional change in this commit.

File: bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java (hunk 3 above)
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java]: Ok - this is confusing as I was expecting a database change as per your PR description, but in reality there are a couple of changes in the same PR?

Comment (mleegwt) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java]: Yeah. Sorry about that. Surprised me as well. The PR was submitted with just the database change. The refactoring was performed later. Git aparently presents all changes on the master in a PR, not just those present at the time of creating the PR. My bad?!

Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java]: Ok - no probs - just a bit confusing (hence my stupid questions!). I’ll take a look tomorrow, but it’s probably ok from my quick look.

Thanks.

File: bundles/binding/org.openhab.binding.zwave/database/zwaveme/zme_blindscontroller.xml [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+			<Item>
+				<Value>1</Value>
+				<Label lang="en">Open blinds</Label>
+			</Item>
+			<Label lang="en">Action on Auto open or close function</Label>
+			<Help lang="en">Defines the action to perform upon auto open or auto close.</Help>
+		</Parameter>
+	</Configuration>
+
+	<Associations>
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/database/zwaveme/zme_blindscontroller.xml]: None of the association groups are configured to send updates to the controller. It seems that at least group 3 looks like it should be if you want to know if the position changes - I'm not familiar with the device though so I might be wrong?

Comment (mleegwt) [inline bundles/binding/org.openhab.binding.zwave/database/zwaveme/zme_blindscontroller.xml]: Group 3 does result in position updates. Does that mean the configuration file should be different? Thanks.

Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/database/zwaveme/zme_blindscontroller.xml]: Yes - you need to add the < SetToController > tag with value true in this group.

https://github.com/cdjackson/HABmin2/wiki/Z-Wave-Product-Database#association-configuration <https://github.com/cdjackson/HABmin2/wiki/Z-Wave-Product-Database#association-configuration>

Comment (mleegwt) [inline bundles/binding/org.openhab.binding.zwave/database/zwaveme/zme_blindscontroller.xml]: Right. Missed the description of the database. Will have a look at it tonight (probably).

Comment (mleegwt) [inline bundles/binding/org.openhab.binding.zwave/database/zwaveme/zme_blindscontroller.xml]: Checked the documentation mentioned above. Fixed setting the controller in the commit below.

**Pull request merged**